nx-X11/programs/Xserver/hw/nxagent/Reconnect.c

```diff
@@ -438,22 +438,10 @@ Bool nxagentReconnectSession(void)
 
   if (nxagentKeyboard != NULL)
   {
-    int size;
+    nxagentOldKeyboard = strndup(nxagentKeyboard, strlen(nxagentKeyboard));
+    /* FIXME: check if nxagentOldKeyboard == NULL and abort */
```

Comment:
I have just tried to add another failedStep and then saw, that this keyboard stuff is done before the first failedStep code. Inbetween there are further initialisations which are referred in  `nxagentReconnectError` so we cannot simply jump there.